File: springapp/src/main/java/com/examly/springapp/service/FoodService.java
package com.examly.springapp.service;

import com.examly.springapp.model.Food;
import java.util.Optional;

import org.springframework.data.domain.Page;

import java.util.List;

public interface FoodService {

    Food addFood(Food food);

    Optional<Food> getFoodById(int foodId);

    // List<Food> getAllFoods();
    Page<Food> getPaginatedFoods(int page, int size);

    Food updateFood(int foodId, Food food);

    boolean deleteFood(int foodId);

    List<Food> getFoodsByUserId(long userId);

}

File: angularapp/src/app/services/food.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Food } from '../models/food.model';
import { API_URL } from '../app.constant';

@Injectable({
  providedIn: 'root'
})
export class FoodService {

  constructor(private client:HttpClient) { }

  getAllFoods():Observable<any>{
    return this.client.get(`${API_URL}/food`);
  }
  
  getPaginatedFoods(page: number, size: number): Observable<any> {
    return this.client.get(`${API_URL}/food?page=${page}&size=${size}`);
  }


  getFoodById(foodId:number):Observable<Food>{
    return this.client.get(`${API_URL}/food/${foodId}`);
  }

  addFood(foodItem:Food):Observable<Food>{
    return this.client.post(`${API_URL}/food`,foodItem);
  }

  updateFood(foodId:number,foodItem:Food):Observable<Food>{
    return this.client.put(`${API_URL}/food/${foodId}`,foodItem);
  }
   
  deleteFood(foodId:number):Observable<any>{
    return this.client.delete(`${API_URL}/food/${foodId}`);
  }




}
